syntax = "proto3";

message RegisterRequest_ {
  string server_name = 1;
  int32 server_port = 2;
  repeated int32 keys = 3;
}

message RegisterResponse_ {
  int32 result = 1;
}

message QueryRequest_ {
  int32 key = 1;
}

message QueryResponse_ {
  string server_name = 1;
  int32 server_port = 2;
}

message TerminateRequest_ {}

message TerminateResponse_ {
  int32 num_keys = 1;
}

service Integration {
  rpc Register(RegisterRequest_) returns (RegisterResponse_);
  rpc Query(QueryRequest_) returns (QueryResponse_);
  rpc Terminate(TerminateRequest_) returns (TerminateResponse_);
}
